core/test/financial_templates/PricelessPositionManager.js

```diff
@@ -593,5 +601,141 @@ contract("PricelessPositionManager", function(accounts) {
+    const tokenHolderFinalCollateral = await collateral.balanceOf(tokenHolder);
+    const tokenHolderFinalSynthetic = await tokenCurrency.balanceOf(tokenHolder);
+
+    // The token holder should gain the value of their synthetic tokens in underlying.
+    // The value in underlying is the number of tokens they held in the beginning * settlement price as TRV
+    // When redeeming 25 tokens at a price of 1.2 we expect to receive 30 collateral tokens (25 * 1.2)
+    const expectedTokenHolderFinalCollateral = toWei("30");
+    assert.equal(tokenHolderFinalCollateral.sub(tokenHolderInitialCollateral), expectedTokenHolderFinalCollateral);
+
+    // The token holder should have no synthetic positions left after settlement.
+    assert.equal(tokenHolderFinalSynthetic, 0);
+
+    //Check the event returned the correct values
+    truffleAssert.eventEmitted(settleExpiredResult, "SettleExpiredPosition", ev => {
+      return (
+        ev.caller == tokenHolder &&
+        ev.collateralAmountReturned == tokenHolderFinalCollateral.sub(tokenHolderInitialCollateral).toString() &&
+        ev.tokensBurned == tokenHolderInitialSynthetic.toString()
+      );
+    });
+
+    // For the sponsor, they are entitled to the underlying value of their remaining synthetic tokens + the excess collateral
+    // in their position at time of settlement - final fees. The sponsor had 100 units of collateral in their position and the final TRV
+    // of their synthetics they sold is 60. Their redeemed amount for this excess collateral is the difference between the two.
+    // The sponsor also has 25 synthetic tokens that they did not sell. This makes their expected redemption = 100 - 60 + 25 * 1.2 - 1 = 69
+    const sponsorInitialCollateral = await collateral.balanceOf(sponsor);
+    const sponsorInitialSynthetic = await tokenCurrency.balanceOf(sponsor);
+
+    // Approve tokens to be moved by the contract and execute the settlement.
+    await tokenCurrency.approve(pricelessPositionManager.address, sponsorInitialSynthetic, {
+      from: sponsor
+    });
+    await pricelessPositionManager.settleExpired({ from: sponsor });
+    const sponsorFinalCollateral = await collateral.balanceOf(sponsor);
+    const sponsorFinalSynthetic = await tokenCurrency.balanceOf(sponsor);
+
+    // The token Sponsor should gain the value of their synthetics in underlying
+    // + their excess collateral from the over collateralization in their position
+    // Excess collateral = 100 - 50 * 1.2 - 1 = 39
+    const expectedSponsorCollateralUnderlying = toBN(toWei("39"));
+    // Value of remaining synthetic tokens = 25 * 1.2 = 30
+    const expectedSponsorCollateralSynthetic = toBN(toWei("30"));
+    const expectedTotalSponsorCollateralReturned = expectedSponsorCollateralUnderlying.add(
+      expectedSponsorCollateralSynthetic
+    );
+    assert(sponsorFinalCollateral.sub(sponsorInitialCollateral), expectedTotalSponsorCollateralReturned);
```

Comment:
Like the other spot, we should probably be using `assert.equal()` here.

Reply:
Just to resolve our earlier conversation, the syntax for assert is:
```
assert(expression, message);
```

So we're effectively just passing expectedTotalSponsorCollateralReturned as the error message, not the thing to compare it to.

Reply:
ah i missed this one